scan/spec/xcpretty_reporter_options_generator_spec.rb

```diff
@@ -0,0 +1,184 @@
+describe Scan do
+  describe Scan::XCPrettyReporterOptionsGenerator do
+    before(:all) do
+      options = { project: "./scan/examples/standard/app.xcodeproj" }
+      Scan.config = FastlaneCore::Configuration.create(Scan::Options.available_options, options)
+    end
+
+    describe "xcpretty reporter options generation" do
+      it "generates options for the junit tempfile report required by scan" do
+        generator = Scan::XCPrettyReporterOptionsGenerator.new(false, "html", "report.html", "test_output", false)
+        reporter_options = generator.generate_reporter_options
+        temp_junit_report = Scan.cache[:temp_junit_report]
+
+        expect(temp_junit_report).not_to be_nil
+        expect(reporter_options).to include("--report junit")
+        expect(reporter_options).to include("--output #{temp_junit_report}")
+      end
+
+      it "generates options for a custom junit report with default file name" do
```

Comment:
I wasn't sure the best way to approach writing this test, since `XCPrettyReporterOptionsGenerator` never had knowledge of `Scan.config`. `test_command_generator.rb` itself contains [the logic](https://github.com/fastlane/fastlane/pull/7365/files#diff-ef49192ad6b55e081d8efc1ddabd1ee8R106) for defaulting to `custom_report_file_name`

I added a class method, `generate_from_scan_config`, to be used by that unit test. See [here](https://github.com/fastlane/fastlane/pull/7365/files#diff-8725c22c4c270712c2f08f44769bdd77R5).